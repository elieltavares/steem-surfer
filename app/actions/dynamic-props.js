import {
  getDynamicGlobalProperties,
  getFeedHistory,
  getRewardFund
} from '../backend/steem-client';
import parseToken from '../utils/parse-token';

export const FETCHED = 'dynamic-props/FETCHED';

export const fetchGlobalProps = () => async dispatch => {
  let globalDynamic;
  let feedHistory;
  let rewardFund;

  try {
    globalDynamic = await getDynamicGlobalProperties();
    feedHistory = await getFeedHistory();
    rewardFund = await getRewardFund();
  } catch (e) {
    return;
  }

  const steemPerMVests =
    (parseToken(globalDynamic.total_vesting_fund_steem) /
      parseToken(globalDynamic.total_vesting_shares)) *
    1e6;
  const base = parseToken(feedHistory.current_median_history.base);
  const quote = parseToken(feedHistory.current_median_history.quote);
  const fundRecentClaims = rewardFund.recent_claims;
  const fundRewardBalance = parseToken(rewardFund.reward_balance);

  dispatch(fetched(steemPerMVests, base, quote, fundRecentClaims, fundRewardBalance));
};

/* action creators */

export const fetched = (
  steemPerMVests,
  base,
  quote,
  fundRecentClaims,
  fundRewardBalance
) => ({
  type: FETCHED,
  payload: { steemPerMVests, base, quote, fundRecentClaims, fundRewardBalance }
});
