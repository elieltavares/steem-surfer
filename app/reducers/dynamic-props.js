import { FETCHED } from '../actions/dynamic-props';

const defaultState = {
  steemPerMVests: 1,
  base: 1,
  quote: 1,
  fundRecentClaims: 1,
  fundRewardBalance: 1
};

export default function dynamicProps(state = defaultState, action) {
  switch (action.type) {
    case FETCHED: {
      const { payload } = action;
      const {
        steemPerMVests,
        base,
        quote,
        fundRecentClaims,
        fundRewardBalance
      } = payload;
      return Object.assign({}, state, {
        steemPerMVests,
        base,
        quote,
        fundRecentClaims,
        fundRewardBalance
      });
    }
    default:
      return state;
  }
}
