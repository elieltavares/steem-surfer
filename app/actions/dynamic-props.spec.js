/* eslint-disable */
import * as actions from './dynamic-props';

describe('dynamic props actions', () => {
  it('(1) fetched action creator', () => {
    const steemPerMVests = 495.05469644322403;
    const base = 0.86;
    const quote = 1;
    const fundRecentClaims = '497885072157522967';
    const fundRewardBalance = 834665.854;

    expect(
      actions.fetched(
        steemPerMVests,
        base,
        quote,
        fundRecentClaims,
        fundRewardBalance
      )
    ).toMatchSnapshot();
  });
});
