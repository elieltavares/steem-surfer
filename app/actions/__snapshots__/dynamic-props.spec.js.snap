// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic props actions (1) fetched action creator 1`] = `
Object {
  "payload": Object {
    "base": 0.86,
    "fundRecentClaims": "497885072157522967",
    "fundRewardBalance": 834665.854,
    "quote": 1,
    "steemPerMVests": 495.05469644322403,
  },
  "type": "dynamic-props/FETCHED",
}
`;
